zerver/tests/test_realm_linkifiers.py

```diff
@@ -114,3 +116,55 @@ def test_delete(self) -> None:
         result = self.client_delete(f"/json/realm/filters/{linkifier_id}")
         self.assert_json_success(result)
         self.assertEqual(RealmFilter.objects.count(), linkifiers_count - 1)
+
+    def test_update(self) -> None:
+        self.login("iago")
+        realm = get_realm("zulip")
+        linkifier_id = do_add_linkifier(
+            realm, "#(?P<id>[123])", "https://realm.com/my_realm_filter/pull/%(id)s"
+        )
+        result = self.client_get("/json/realm/linkifiers")
+        self.assert_json_success(result)
+
+        do_update_linkifier(
+            realm,
+            linkifier_id,
+            "#(?P<id>[0-9]+)",
+            "https://realm.com/my_realm_filter/issues/%(id)s",
+        )
+        result = self.client_get("/json/realm/linkifiers")
+        self.assert_json_success(result)
+        # Verify that the linkifier is updated accordingly.
+        linkifier = result.json()["linkifiers"]
+        self.assertEqual(len(linkifier), 1)
+        self.assertEqual(linkifier[0]["pattern"], "#(?P<id>[0-9]+)")
+        self.assertEqual(
+            linkifier[0]["url_format"], "https://realm.com/my_realm_filter/issues/%(id)s"
+        )
+
+        data = {
+            "pattern": r"ZUL-(?P<id>\d++)",
+            "url_format_string": "https://realm.com/my_realm_filter/%(id)s",
+        }
+        result = self.client_patch(f"/json/realm/filters/{linkifier_id}", info=data)
+        self.assert_json_error(
+            result, "Invalid linkifier pattern.  Valid characters are [ a-zA-Z_#=/:+!-]."
+        )
+
+        data["pattern"] = r"ZUL-(?P<id>\d+)"
+        data["url_format_string"] = "$fgfg"
+        result = self.client_patch(f"/json/realm/filters/{linkifier_id}", info=data)
+        self.assert_json_error(result, "Enter a valid URL.")
+
+        data["pattern"] = r"PR (?P<id>[0-9]+)"
+        data[
+            "url_format_string"
+        ] = "https://example.com/~user/web#view_type=type&model=model&action=12345&id=%(id)s"
+        result = self.client_patch(f"/json/realm/filters/{linkifier_id}", info=data)
+        self.assert_json_success(result)
+        self.assertIsNotNone(re.match(data["pattern"], "PR 123"))
```

Comment:
Doing an `assert_json_success` implies that the request was made successfully or not.
`assertIsNotNone` is done to verify that the `notify_linkifiers` function (in `actions.py`) sends an event containing the updated linkifiers that we cross-check in `assertIsNotNone`.

Reply:
Sorry for the confusion here. What I meant was that, once you have addressed https://github.com/zulip/zulip/pull/17045#discussion_r613996007, you can club that and this together, because they both test a successful linkifier update.

Reply:
Got it! That will help to reduce these duplicate tests when https://github.com/zulip/zulip/pull/17045#discussion_r613996007 is addressed. Thanks!